README.md

```diff
@@ -1,80 +1,134 @@
+    ![Overview](./ReadmeFiles/topology.png)
 
-In this application, the platform specific projects are effectively only responsible from presenting the UI. Each application receives an input search term from the user, and calls a method `SearchByAlias` in the application's `DirectorySearcherLib` PCL.  `SearchByAlias` returns a `User` object, which is used to present information about the search results such as the user's name, user principal name, and phone number.
+### Scenario
 
-The `DirectorySearcherLib` PCL is the application's shared C# code base, which contains both the identity related logic and the search related logic.  It leverages ADAL.NET v3, which is also a PCL, to automatically sign users in with the OAuth 2.0 protocol, acquire an access token for the AAD Graph API, cache tokens, maintain user sessions, and so on.  It then queries AAD's Graph API for information about a user in the authenticated user's tenant with a matching alias, and returns results to the platform specific UI code.  By writing both the identity and search logic in the `DirectorySearcherLib` PCL, the code only needs to be written once and can be reused across each platform.
+The user enters an alias in the organization of interest an presses search. If needed, the user is asked to sign in to that organization and to consent for the application to read user's basic profile. 
+Then, if the alias is found in the user's organization, the profile of the corresponding user is written in the UI. This profile consists in first name, last name, email address, and phone number. The picture below shows the UI for a UWP application, but it's really the same for other platforms.
 
-## How To Run The Sample
+![UI](./ReadmeFiles/ui.png)
+
+## How to run the sample
+
+To run this entire sample, you'll need:
 
-To run this entire sample you will need:
 - An Internet connection
 - A Windows machine (necessary if you want to run the Windows Store or WPF apps)
 - An OS X machine (necessary if you want to run the iOS app)
 - Visual Studio 2017 (recommended) which comes with Xamarin for iOS, Android, and Universal platform;
 - An Azure subscription (a free trial is sufficient)
 
 Every Azure subscription has an associated Azure Active Directory tenant.  If you don't already have an Azure subscription, you can get a free subscription by signing up at [https://azure.microsoft.com](https://azure.microsoft.com).  All of the Azure AD features used by this sample are available free of charge.
+This sample will not work with a Microsoft account (formerly Windows Live account). Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a Microsoft account and have never created a user account in your directory before, you need to do that now.

```

Comment:
>Microsoft account [](start = 167, length = 17)

Minor:
Some places it says Microsoft account. Other places it says Microsoft Account.